File: backend/app/schemas/error_log.py
# Path: app/schemas/error_log.py
# File: error_log.py
# Created: 2026-04-09
# Purpose: Pydantic schemas for error log create/read
# Caller: app/routers/errors.py
# Callees: pydantic
# Data In: HTTP request bodies
# Data Out: Validated error log data
# Last Modified: 2026-04-09

from datetime import datetime

from pydantic import BaseModel


class ErrorLogCreate(BaseModel):
    project_id: int | None = None
    agent_id: int | None = None
    source: str = "frontend"
    endpoint: str | None = None
    error_type: str | None = None
    message: str
    stack_trace: str | None = None
    file_path: str | None = None
    function_name: str | None = None
    line_number: int | None = None
    status_code: int | None = None


class ErrorLogRead(BaseModel):
    id: int
    project_id: int | None
    agent_id: int | None
    source: str
    endpoint: str | None
    error_type: str | None
    message: str
    stack_trace: str | None
    file_path: str | None
    function_name: str | None
    line_number: int | None
    status_code: int | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/helpContent/sections/error_log.js
// Path: src/helpContent/sections/error_log.js
// File: error_log.js
// Created: 2026-06-25
// Purpose: Help content for the system-wide Error Log view (DWB-473). Covers the
//          /errors page (source/project filters, stack-trace expansion) plus the
//          client error logging endpoints (errors + client-logs telemetry feed).
// Caller: helpContent/index.js (auto-discovered via import.meta.glob)
// Callees: none (plain data)
// Data In: none
// Data Out: default export: one help section object
// Last Modified: 2026-06-25

export default {
  key: 'error_log',
  title: 'Error Log',
  summary: {
    why: 'One place to see failures across the whole system: backend exceptions, frontend crashes, and hook failures from every project.',
    how: 'Scan the most-recent-first list, filter by source or project, and click a row to expand its stack trace.',
    where: 'The error_log link in the Overview nav (the /errors route).',
  },
  bullets: [
    'The list auto-refreshes every 10 seconds; hit "$ refresh" to pull immediately.',
    'Each row is tagged with its source: [BE] backend, [FE] frontend, or [HK] hook, plus the project prefix, HTTP status code (500s highlighted), endpoint, and message.',
    'Filter by source and by active project; the two filters compose. Click any row to expand the error type, origin (file:function:line), and full stack trace.',
    'Errors are written via POST /api/errors with a source of backend, frontend, or hook; query them with GET /api/errors using project_id, source, and limit (default 50, capped at 200).',
    'A separate frontend telemetry feed batches lower-severity logs to POST /api/client-logs under a lenient never-5xx contract: malformed records drop individually and the rest still land.',
    'Each client log carries a level (debug, info, warn, error), a category, and an optional route; query them with GET /api/client-logs by level, category, route, or since.',
    'Hook scripts always exit 0, so a logged hook error never blocks the caller.',
  ],
};
